src/wovnio/html/HtmlConverter.php

```diff
@@ -48,64 +48,39 @@ public function __construct($html, $encoding, $token, $store = null, $headers =
    *
    * @return array converted html and HtmlReplaceMarker
    */
-  public function convertToAppropriateForApiBody($removeParts = true)
+  public function convertToAppropriateForApiBody()
   {
-    if ($this->encoding && in_array($this->encoding, self::$supported_encodings)) {
-      $encoding = $this->encoding;
-    } else {
-      // Encoding detection uses 30% of execution time for this method.
-      $encoding = mb_detect_encoding($this->html, self::$supported_encodings);
-    }
-
-    $dom = simple_html_dom::str_get_html($this->html, $encoding, false, false, $encoding, false);
-    $this->insertSnippet($dom);
-    if (isset($this->store) && isset($this->headers)) {
-      $this->insertHreflangTags($dom);
-    }
-
-    $marker = new HtmlReplaceMarker();
-    if ($removeParts) {
-      $this->removeWovnIgnore($dom, $marker);
-      $this->removeForm($dom, $marker);
-      $this->removeScript($dom, $marker);
-    }
-
-    $converted_html = $dom->save();
+    $this->html = $this->insertSnippet($this->html);
+    $this->html = $this->insertHreflangTags($this->html);
 
-    // Without clear(), Segmentation fault will be raised.
-    // @see https://sourceforge.net/p/simplehtmldom/bugs/103/
-    $dom->clear();
-    unset($dom);
-
-    if ($removeParts) {
-      $converted_html = $this->removeBackendWovnIgnoreComment($converted_html, $marker);
-    }
-    return array($converted_html, $marker);
+    return array($this->html);
   }
 
   /**
    * Insert wovn's snippet to ensure snippet is always inserted.
    * When snippet is always inserted, do nothing
    *
-   * @param simple_html_dom_node $dom
+   * @param simple_html_dom_node $html
    */
-  private function insertSnippet($dom)
+  private function insertSnippet($html)
   {
-    foreach ($dom->find('script') as $node) {
-      if (strpos($node->getAttribute('src'), '//j.wovn.io/') !== false ||
-        strpos($node->getAttribute('src'), '//j.dev-wovn.io:3000/') !== false) {
-        return;
-      }
+    $snippet_regex = "<script(.*)src='(j.wovn.io|j.dev-wovn.io)'(.*)><\/script>";
+    if (preg_match("/$snippet_regex/i", $html)) {
+      return;
     }
 
     $token = $this->token;
-    $insert_tags = array('head', 'body', 'html');
-    foreach ($insert_tags as $tag_name) {
-      $parents = $dom->find($tag_name);
-      if (count($parents) > 0) {
-        $parent = $parents[0];
-        $parent->innertext = "<script src='//j.wovn.io/1' data-wovnio='key=$token' data-wovnio-type='backend_without_api' async></script>" . $parent->innertext;
-        return;
+    $snippet_code = "<script src='//j.wovn.io/1' data-wovnio='key=$token' data-wovnio-type='backend_without_api' async></script>";
+    $parent_tags = array('<head>', '<body>', '<html>');
```

Comment:
any tag can have some attributes so we should allow

Reply:
You're right. I changed code to cover that. Thanks